### DISCUSSION TITLE
Pitch Correction

### DISCUSSION THREAD

**Discussion by 47hao** [playcanvas/engine#2715] [Q&A]
Hi! I'm looking to modify the playcanvas engine such that the set pitch function only changes the playback rate of sounds, and keeps its pitch, so I can speed up and slow down sounds without changing their pitches.

I saw documentation here that says playbackRate pitch corrects audio by default, but am not sure whether this is the implementation playcanvas uses.
https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/playbackRate

Accordingly I've gone through the playcanvas source code, commenting/disabling code that changes the pitch and keeping the changes to playbackrate.

I've also found that changing _pitch and pitch variables does not seem to have an effect on the actual sound pitch.

Any pointers/helpful information on this would be appreciated!

**Comment by willeastcott** [comment]
@vkalpias wrote the audio subsystems. Any thoughts on this?

**Reply by 47hao** [reply]
A little more digging makes it look more like playcanvas uses the AudioBufferSourceNode interface instead. Is this the case?

**Reply by vkalpias** [reply]
Hi - PlayCanvas uses AudioBufferSourceNode internally if it's supported otherwise it falls back to an `<audio>` element.

**Reply by vkalpias** [reply]
> @vkalpias wrote the audio subsystems. Any thoughts on this?

Changing the pitch variable ends up changing the playbackRate of the audio source internally. Maybe `pitch` was the wrong name to use for this?
